components/renderActivities.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import ProcessCard from "./ProcessCard";
+import EconomicEventCard from "./EconomicEventCard";
+
+
+const RenderActivities = ({userActivity}:{userActivity:any}) => {
+  const obj = userActivity;
+  const isProcess = () => obj.__typename == "Process"
+  const isEconomicEvent = () => obj.__typename == "EconomicEvent"
+
+  if (!obj)
+    return (<><b>nothing to show</b><br /><br /></>)
```

Comment:
We should really start to look at `<Suspend />` and `swr` bonded to something like: https://skeletonreact.com/